pkg/controller/che/che_controller.go

```diff
@@ -255,6 +255,14 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		logrus.Errorf("An error occurred when detecting current infra: %s", err)
 	}
 
+	// Check Che CR correctness
+	if err := ValidateCheCR(instance, isOpenShift); err != nil {
+		// Che cannot be deployed with current configuration.
+		// Print error message in logs  and wait until the configuration is changed.
+		logrus.Error(err)
+		return reconcile.Result{}, nil
```

Comment:
return reconcile.Result{}, err

Reply:
Have you tested if Eclipse Che is continued deploying when CR changed and validation passed?

Reply:
No, because in that case it will mean requeue reconcile request and operator will fall into infinite loop.